pkg/manager/manager_deep_test.go

```diff
@@ -0,0 +1,279 @@
+		).Return(&gnmi.SetResponse{
+			Response: []*gnmi.UpdateResult{
+				{Path: changePath1},
+			},
+			Timestamp: time.Now().Unix(),
+		}, nil).AnyTimes()
+
+		return southbound.TargetIf(mockTarget)
+	}
+
+	mgrTest.Run(mockTargetCreationfn)
+
+	time.Sleep(time.Millisecond * 100)
+
+	assert.Assert(t, networkChange1 != nil)
+	err = mgrTest.NetworkChangesStore.Create(networkChange1)
+	assert.NilError(t, err, "Unexpected failure when creating new NetworkChange")
+	time.Sleep(time.Millisecond * 10)
+
+	return mgrTest, &AllMocks{
+		MockStores: &mockstore.MockStores{
+			DeviceStore:          mockDeviceStore,
+			NetworkChangesStore:  nil,
+			DeviceChangesStore:   nil,
+			NetworkSnapshotStore: nil,
+			DeviceSnapshotStore:  nil,
+			LeadershipStore:      nil,
+			MastershipStore:      nil,
+		},
+	}
+}
+
+func Test_GetNetworkConfig_Deep(t *testing.T) {
+
+	mgrTest, _ := setUpDeepTest(t)
+
+	result, err := mgrTest.GetTargetConfig(device1, deviceVersion1, "/*", 0)
+	assert.NilError(t, err, "GetTargetConfig error")
+	time.Sleep(time.Millisecond * 100)
+
+	assert.Equal(t, len(result), 1, "Unexpected result element count")
+
+	assert.Equal(t, result[0].Path, test1Cont1ACont2ALeaf2A, "result %s is different")
+}
+
+func Test_SetNetworkConfig_Deep(t *testing.T) {
+	mgrTest, _ := setUpDeepTest(t)
+
+	// First verify the value beforehand
+	originalChange, _ := mgrTest.NetworkChangesStore.Get(networkChange1)
+	assert.Equal(t, len(originalChange.Changes[0].Values), 1)
+	assert.Equal(t, originalChange.Changes[0].Values[0].Path, test1Cont1ACont2ALeaf2A)
+	assert.Equal(t, originalChange.Changes[0].Values[0].Value.Type, devicechange.ValueType_FLOAT)
+	assert.Equal(t, (*devicechange.TypedFloat)(originalChange.Changes[0].Values[0].Value).Float32(), float32(valueLeaf2B159))
+
+	// Making change
+	updates := make(devicechange.TypedValueMap)
+	updates[test1Cont1ACont2ALeaf2A] = devicechange.NewTypedValueUint64(valueLeaf2A789)
+	deletes := []string{test1Cont1ACont2ALeaf2C}
+	updatesForDevice1, deletesForDevice1, deviceInfo := makeDeviceChanges(device1, updates, deletes)
+
+	// Verify the change
+	validationError := mgrTest.ValidateNetworkConfig(device1, deviceVersion1, deviceTypeTd, updates, deletes)
+	assert.NilError(t, validationError, "ValidateTargetConfig error")
+
+	// Set the new change
+	const testNetworkChange networkchange.ID = "Test_SetNetworkConfig"
+	err := mgrTest.SetNetworkConfig(updatesForDevice1, deletesForDevice1, deviceInfo, string(testNetworkChange))
+	assert.NilError(t, err, "SetTargetConfig error")
+	time.Sleep(time.Millisecond * 10) // Have to wait for change to be reconciled
+	testUpdate, _ := mgrTest.NetworkChangesStore.Get(testNetworkChange)
+	assert.Assert(t, testUpdate != nil)
+	assert.Equal(t, testUpdate.ID, testNetworkChange, "Change Ids should correspond")
+	assert.Equal(t, changetypes.Phase_CHANGE, testUpdate.Status.Phase)
+	assert.Equal(t, changetypes.State_COMPLETE, testUpdate.Status.State)
```

Comment:
this is the most essential line in the whole patch - checks that the Network change is complete - i.e. has gone down to device controller and on to synchronizer, made the change (to a mocked target), got its result back and the reconciler marked the `DeviceChange` and the `NetworkChange` complete